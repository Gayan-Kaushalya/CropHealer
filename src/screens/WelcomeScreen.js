import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";
import React from "react";

const WelcomeScreen = ({ navigation }) => {
  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <Image
        source={require("./../../assets/CropHealer Logo.jpg")}
        style={{ width: 200, height: 200 }}
      />

      <Text style={{ color: "green", fontSize: 42, fontWeight: "bold" }}>
        Welcome to CropHealer!
      </Text>
      <Text
        style={{
          fontSize: 24,
          fontWeight: "bold",
          marginTop: 20,
          marginBottom: 20,
          alignContent: "center",
          color: "black",
        }}
      >
        Your AI Powered Plant Disease Identifier
      </Text>

      {/* Add a button to navigate to the Prediction screen */}
      <TouchableOpacity
        onPress={() => navigation.navigate("Predict")}
        style={{
          backgroundColor: "green",
          padding: 10,
          borderRadius: 5,
          width: "80%",
          alignItems: "center",
        }}
      >
        <Text
          style={{
            color: "white",
            fontSize: 20,
            fontWeight: "bold",
            textAlign: "center",
          }}
        >
          Diagnose
        </Text>
      </TouchableOpacity>



      <TouchableOpacity
        onPress={() => navigation.navigate("Article List")}
        style={{
          backgroundColor: "green",
          padding: 10,
          borderRadius: 5,
          width: "80%",
          alignItems: "center",
          marginTop: 10,
        }}
      >
        <Text
          style={{
            color: "white",
            fontSize: 20,
            fontWeight: "bold",
            textAlign: "center",
          }}
        >
          Explore
        </Text>
      </TouchableOpacity>

      <TouchableOpacity
        onPress={() => navigation.navigate("Login")}
        style={{
          backgroundColor: "green",
          padding: 10,
          borderRadius: 5,
          width: "80%",
          alignItems: "center",
          marginTop: 10,
        }}
      >
        <Text
          style={{
            color: "white",
            fontSize: 20,
            fontWeight: "bold",
            textAlign: "center",
          }}
        >
          Login
        </Text>
      </TouchableOpacity>
    </View>
  );
};

export default WelcomeScreen;

const styles = StyleSheet.create({});
