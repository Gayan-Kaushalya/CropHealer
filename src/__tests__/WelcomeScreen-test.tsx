import React from "react";
import { render } from "@testing-library/react-native";
import WelcomeScreen from "../screens/WelcomeScreen"; // Adjust the path as necessary
import { NavigationContainer } from "@react-navigation/native";

// Mocking the navigation object
const mockNavigate = jest.fn();

describe("<WelcomeScreen />", () => {
  it("renders correctly", () => {
    const { toJSON } = render(
      <NavigationContainer>
        <WelcomeScreen navigation={{ navigate: mockNavigate }} />
      </NavigationContainer>
    );

    // Create a snapshot of the rendered component
    expect(toJSON()).toMatchSnapshot();
  });
});
