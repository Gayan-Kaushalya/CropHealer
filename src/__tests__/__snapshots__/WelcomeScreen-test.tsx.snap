// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
  />
  <Text
    style={
      {
        "color": "green",
        "fontSize": 42,
        "fontWeight": "bold",
      }
    }
  >
    Welcome to CropHealer!
  </Text>
  <Text
    style={
      {
        "alignContent": "center",
        "color": "black",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    Your AI Powered Plant Disease Identifier
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 5,
        "opacity": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Diagnose
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 5,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Explore
    </Text>
  </View>
</View>
`;
